### PR TITLE
Fix model-archiver to fail if extra files are missing

### PR DESCRIPTION
**Pull request by mreso**
## Description

This PR lets the model-archiver fail if a file given in the extra-file list does not exist.
Additionally, it refactors the integration test to not start a subprocess to determine code coverage.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] pytest model_archiver/tests/integ_tests/
```
=============================================================================== test session starts ===============================================================================
platform linux -- Python 3.9.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/ubuntu/serve/model-archiver
plugins: mock-3.10.0, cov-4.0.0
collected 3 items

model_archiver/tests/integ_tests/test_integration_model_archiver.py ...                                                                                                     [100%]

================================================================================ 3 passed in 0.09s ================================================================================
```
- [x] pytest model_archiver/tests/unit_tests/
```
=============================================================================== test session starts ===============================================================================
platform linux -- Python 3.9.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/ubuntu/serve/model-archiver
plugins: mock-3.10.0, cov-4.0.0
collected 28 items

model_archiver/tests/unit_tests/test_model_packaging.py ....                                                                                                                [ 14%]
model_archiver/tests/unit_tests/test_model_packaging_utils.py .......................                                                                                       [ 96%]
model_archiver/tests/unit_tests/test_version.py .                                                                                                                           [100%]

=============================================================================== 28 passed in 0.10s ================================================================================
```

## Checklist:

- [X] Did you have fun?
- [X] Have you added tests that prove your fix is effective or that this feature works?
- [X] Has code been commented, particularly in hard-to-understand areas?
- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: model-archiver/model_archiver/tests/integ_tests/test_integration_model_archiver.py [hunk 1]
```diff
@@ -40,16 +40,28 @@ def delete_file_path(path):
         pass
 
 
-def run_test(test, cmd):
-    it = test.get("iterations") if test.get("iterations") is not None else 1
+def run_test(test, args, mocker):
```
Comment (msaroufim) [inline model-archiver/model_archiver/tests/integ_tests/test_integration_model_archiver.py:43]: do you mind giving the rest of the team a brief presentation on test mocking and pytest more generally? I'd for one certainly really enjoy it

**Review by msaroufim — APPROVED** [review]

File: model-archiver/model_archiver/tests/integ_tests/test_integration_model_archiver.py (hunk 1 above)
Comment (mreso) [reply model-archiver/model_archiver/tests/integ_tests/test_integration_model_archiver.py:43]: Happy to do so

**Review by agunapal — APPROVED** [review]

File: model-archiver/model_archiver/model_packaging_utils.py [hunk 2]
```diff
@@ -164,6 +165,10 @@ def copy_artifacts(model_name, **kwargs):
 
                 if file_type == "extra_files":
                     for path_or_wildcard in path.split(","):
+                        if not Path(path_or_wildcard).exists():
```
Comment (otakudj) [inline model-archiver/model_archiver/model_packaging_utils.py:168]: this "if" lead [wildcard feature](https://github.com/pytorch/serve/pull/2142) invalid @mreso

Comment (mreso) [reply model-archiver/model_archiver/model_packaging_utils.py:168]: good catch @otakudj created #3304 to fix

**Pull request merged**